driver/src/main/java/org/neo4j/driver/internal/util/AddressUtil.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.util;
+
+import java.util.regex.Pattern;
+
+public class AddressUtil
+{
+    private static final Pattern LOCALHOST = Pattern.compile( "^(localhost|127(\\.\\d+){3})$", Pattern.CASE_INSENSITIVE );
+
+    /**
+     * Return true if the host provided matches "localhost" or "127.x.x.x".
+     *
+     * @param host the host name to test
+     * @return true if localhost, false otherwise
+     */
+    public static boolean isLocalhost( String host )
+    {
+        return LOCALHOST.matcher( host ).matches();
```

Comment:
This is not a good way to check for localhost. There can be multiple network cards, interfaces and what not. Plus regex matches things like `127.99999.99.123123123123`. Class `InetAddress` has methods like `#isAnyLocalAddress()`, `#isLoopbackAddress()`. It seems to be useful to check if address is defined on any interface like this `NetworkInterface.getByInetAddress(address) != null`. I guess we can adopt smth similar to http://stackoverflow.com/a/2406819/5517896